packages/samples/headless-react/src/components/result-list/result-list.class.tsx

```diff
@@ -0,0 +1,70 @@
+import {Component} from 'react';
+import {
+  buildResultList,
+  ResultList as ResultListController,
+  ResultListState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+export class ResultList extends Component {
+  private readonly controller: ResultListController;
+  public state: ResultListState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: {}) {
+    super(props);
+
+    this.controller = buildResultList(engine);
+    this.state = this.controller.state;
+  }
+
+  componentDidMount() {
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private showMore() {
+    this.controller.fetchMoreResults();
+  }
+
+  private get resultsAvailable() {
+    return this.state.results.length > 0;
+  }
+
+  private get results() {
+    return this.state.results.map((result) => (
+      <li key={result.uniqueId}>
+        <article>
+          <h2>
+            <a href={result.clickUri}>{result.title}</a>
+          </h2>
+          <p>{result.excerpt}</p>
+        </article>
+      </li>
+    ));
+  }
+
+  render() {
+    return (
+      <div>
+        <ul>{this.resultsAvailable ? this.results : <li>No results</li>}</ul>
+        {this.resultsAvailable && (
+          <button
+            onClick={() => this.showMore()}
+            disabled={this.state.isLoading}
+          >
+            Show more
+          </button>
```

Comment:
I would vote to remove the `Show More` button to simplify the sample. I see the samples as reducing the controller to it's essence, focused on the key functionality versus all possibilities.

Think of the person looking at the sample as someone learning about Coveo and Headless for the first time. We want to show how easy it is to get started. Part of this is keeping the lines of code to a minimum while still being useful.
It also helps to have all rendered elements inside the `render` function instead of needing to scroll up to the methods.

I would suggest:

```
render() {
    if (!this.state.results.length) {
        return <div>No results</<div>
    }

    return (
    <div>
     {this.state.results.map((result) => (
      <li key={result.uniqueId}>
        <article>
          <h2>
            <a href={result.clickUri}>{result.title}</a>
          </h2>
          <p>{result.excerpt}</p>
        </article>
      </li>
    ));}
    </div>
}
```

As a first-time viewer, I can see what is rendered in one glance.

Reply:
Good point, I'll remove the "show more" button and move all rendering into the render function.